FV3/coarse_graining/coarse_graining.F90

```diff
@@ -1065,4 +1140,45 @@ subroutine eddy_covariance_3d_weights(weights, field_a, field_b, coarse)
    call anomaly_3d_weights_3d_array(weights, field_b, anom_b)
    call weighted_block_average(weights, anom_a * anom_b, coarse)
  end subroutine eddy_covariance_3d_weights
+
+ ! Compute pressure at layer midpoints following Eq. 3.17 of Simmons
+ ! and Burridge (1981), MWR.
+ subroutine compute_pfull_from_phalf(phalf, pfull)
```

Comment:
Just double checking, I see that in `fv_dynamics`, they calculate `pfull` from:
```
      do k=1,npz
         ph1 = ak(k  ) + bk(k  )*flagstruct%p_ref
         ph2 = ak(k+1) + bk(k+1)*flagstruct%p_ref
         pfull(k) = (ph2 - ph1) / log(ph2/ph1)
      enddo
```
but here we want to get the coarsened `pfull` from a previously coarsened `phalf`? So essentially we replace `ph` with `phalf`.

Reply:
Yup, exactly.